src/features/txnDelegatation/SignHotspot.tsx

```diff
@@ -0,0 +1,271 @@
+          owner: ownerKeypair,
+          payer: ownerIsPayer ? ownerKeypair : undefined,
+        })
+        let finalTxn = txnOwnerSigned.toString()
+        if (!ownerIsPayer) {
+          const stakingServerSignedTxn = await getStakingSignedTransaction(
+            locationTxn.gateway.b58,
+            finalTxn,
+          )
+          finalTxn = stakingServerSignedTxn
+        }
+
+        responseParams.assertTxn = finalTxn
+      }
+
+      callback(responseParams)
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e)
+    }
+  }, [callback, gatewayTxn, locationTxn])
+
+  const handleCancel = useCallback(async () => {
+    callback({ status: 'user_cancelled' })
+  }, [callback])
+
+  const name = useMemo(() => {
+    if (!gatewayTxn?.gateway?.b58 && !locationTxn?.gateway?.b58) return
+    return animalHash(
+      gatewayTxn?.gateway?.b58 || locationTxn?.gateway?.b58 || '',
+    )
+  }, [gatewayTxn?.gateway?.b58, locationTxn?.gateway?.b58])
+
+  const location = useMemo(() => {
+    return locationTxn?.location
+  }, [locationTxn?.location])
+
+  useEffect(() => {
+    if (!parsedToken) return
+    verifyAppLinkAuthToken(parsedToken)
+      .then((valid) => {
+        setValidated(valid)
+      })
+      .catch(() => {
+        setValidated(false)
+      })
+  }, [parsedToken, token])
+
+  return (
+    <SafeAreaBox
+      backgroundColor="primaryBackground"
+      flex={1}
+      padding="xl"
+      justifyContent="center"
+    >
+      <Text variant="bold" fontSize={36}>
+        {t(gatewayTxn ? 'signHotspot.title' : 'signHotspot.titleLocationOnly')}
+      </Text>
+
+      <Box backgroundColor="white" borderRadius="l" padding="l" marginTop="l">
+        <Text variant="regular" fontSize={16} color="purpleText">
+          {t('signHotspot.name')}
+        </Text>
+        <Text
+          variant="bold"
+          fontSize={20}
+          color="primaryBackground"
+          marginBottom="m"
+        >
+          {name}
+        </Text>
+        {location && (
+          <>
+            <Text variant="regular" fontSize={16} color="purpleText">
+              {t('signHotspot.location')}
+            </Text>
+            <Text
+              variant="bold"
+              fontSize={20}
+              color="primaryBackground"
+              marginBottom="m"
+            >
+              {location}
+            </Text>
+          </>
+        )}
+        <Box flexDirection="row">
+          {locationTxn?.gain && (
+            <Box marginEnd="xxl">
+              <Text variant="regular" fontSize={16} color="purpleText">
+                {t('signHotspot.gain')}
+              </Text>
+              <Text
+                variant="bold"
+                fontSize={20}
+                color="primaryBackground"
+                marginBottom="m"
+              >
+                {locationTxn.gain}
+              </Text>
+            </Box>
+          )}
+          {locationTxn?.elevation && (
```

Comment:
I think elevation can be 0, might want to check `locationTxn?.elevation !== undefined` instead